x-pack/legacy/plugins/siem/public/pages/hosts/host_details.tsx

```diff
@@ -151,6 +151,11 @@ const HostDetailsComponent = pure<HostDetailsComponentProps>(
                             refetch={refetch}
                             setQuery={setQuery}
                             to={to}
+                            narrowDateRange={(min: number, max: number) => {
+                              setTimeout(() => {
+                                setAbsoluteRangeDatePicker({ id: 'global', from: min, to: max });
+                              }, 500);
```

Comment:
Wait a bit here as chart updates the state for itself when `BrushEnd`. If we update the date picker immediately after brushEnd, we may unmount the chart for fetching data before `setState` completed.
![set-title-2](https://user-images.githubusercontent.com/6295984/62314817-83d7e800-b4c6-11e9-8117-ebab6997dea9.gif)

Reply:
I wish I had a better suggestion for this, but FWIW as a user, the delay doesn't feel perceivable.

Reply:
I think this a-ok as is as the UI doesn't feel laggy or anything, but I wonder what other ways we can get around this?

Two that come to mind would be:

* Use `narrowDateRange` to set a flag that can be used as the dependency for a `useEffect` that runs after the state update.
* If there's a way to cancel/prevent said state update from propagating within `narrowDateRange`.

Either way, just thoughts -- doesn't appear to be any need to make any modifications :)

Reply:
Thanks, I didn't think of these options! Those are very interesting ideas, but I haven't had it work yet 😢, I'll follow up the update from charts and improve this accordingly.